packages/apollo-fragment-utils/src/index.ts

```diff
@@ -1,18 +1,29 @@
-import gql from 'graphql-tag';
-import { DocumentNode } from 'graphql';
+import { gql } from '@apollo/client/core';
+import { DocumentNode, Kind } from 'graphql';
 
 export function getFragmentInfo(fragment: string | DocumentNode) {
   const fragmentAST = typeof fragment === `string` ? gql(fragment) : fragment;
-  const fragmentDefinitions =
-    fragmentAST.definitions && fragmentAST.definitions[0];
-  const fragmentName = fragmentDefinitions && fragmentDefinitions.name.value;
-  const fragmentTypeName =
-    fragmentDefinitions && fragmentDefinitions.typeCondition.name.value;
+  // ensure we are working with the right kind of definition, otherwise throw warning
+  if (
+    fragmentAST.definitions &&
+    fragmentAST.definitions[0] &&
+    fragmentAST.definitions[0].kind === Kind.FRAGMENT_DEFINITION
+  ) {
```

Comment:
Had to make some updates here for TS to compile.

Would appreciate feedback on the `console.warn` usage here